src/flatFieldCorrect.cpp

```diff
@@ -0,0 +1,139 @@
+// This file performs flat field correction on the input image, to compensate for uneven illumination on the image plane
+
+// Author Mark Main
+
+#include "flatFieldCorrect.h"
+
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/opencv.hpp>
+
+
+
+
+void flatFieldCorrect::receiveFrame(frame newFrame) {
+    if (!enabled){ 
+        newFrame.setParameter(paramLabel, "OFF");
+        frameCb->receiveFrame(newFrame);
+        return;
+    }
+    // do stuff here
+
+    // passing frame into the edge detection function
+    flatField(newFrame); 
+
+
+}
+
+
+void flatFieldCorrect::updateSettings(std::map<std::string, std::string> metadata){
+    
+    std::string rec = metadata[paramLabel];
+
+    bool desired = (rec == "ON");
+    // std::cout<<rec<<std::endl;
+
+    // std::cout<<desired<<std::endl;
+
+    if (enabled != desired){
+        toggleEnable();
+    }
+
+    
+}
+void flatFieldCorrect::updateAverage(frame f) {
+    // Load reference images
+    std::string pathname = getenv("HOME");
+    pathname += + "/cellUviewGallery/.FlatFieldGallery/";
+    std::vector<cv::Mat> reference_images;
+    for (int i = 1; i <= 20; i++) {
+        std::string flatFieldPath = pathname + "flatField" + std::to_string(i) + ".jpg";
+        cv::Mat reference_image = cv::imread(flatFieldPath);
+        reference_images.push_back(reference_image);
+    }
+
+    // Calculate the average of the reference images
+    cv::Mat average_reference_image;
+    cv::addWeighted(reference_images[0], 1.0/20, reference_images[1], 1.0/20, 0, average_reference_image);
```

Comment:
please move this inside loop and define the numbers. define what 20 is eg int numFlatFieldCaptures = 20. Then use variable throughout

Reply:
Fine, comments added to explain steps. This avoids initialisation of the aveage_reference_image with zeros.